packages/dds/tree/src/feature-libraries/object-forest/objectForest.ts

```diff
@@ -136,11 +136,23 @@ class ObjectForest extends SimpleDependee implements IEditableForest {
 			onMoveOut: (index: number, count: number, id?: Delta.MoveId): void => {
 				const [parent, key] = cursor.getParent();
 				const sourceField = getMapTreeField(parent, key, false);
-				const field = this.detachRangeOfChildren(sourceField, index, index + count);
+				const startIndex = index;
+				const endIndex = index + count;
+				assertValidIndex(startIndex, sourceField, true);
+				assertValidIndex(endIndex, sourceField, true);
+				assert(
+					startIndex <= endIndex,
+					0x371 /* detached range's end must be after its start */,
+				);
+				const newField = sourceField.splice(startIndex, endIndex - startIndex);
+				const field = this.addFieldAsDetached(newField);
 				if (id !== undefined) {
 					moves.set(id, field);
 				} else {
 					this.delete(field);
+					if (sourceField.length === 0) {
```

Comment:
this should be outside the if here: moving all the content out of a field should still delete the source field.

Adding a regression test covering this case would be nice, but I know moves are a bit harder to do at the delta layer. If you can find an easy way to do it, add the test, otherwise just fix this bug.

Reply:
The fix in chunked forest has the same issue. Both of these could just be moved after the else blocks they are in to fix it

Reply:
I added a move test case, as I was able to copy a test case which already uses a `JsonableTree[]` defined as `nestedContent`. Should I update our original test case to use the `nestedContent`, to be consistent with the rest of the file? Or should it stick to the new simplified version of the tree?

Test I copied was from "move out and move in" under "can apply deltas with".